### ISSUE TITLE
Transpiling doesn't raise any error if the `--binary parameter` is incorrect

### ISSUE THREAD

**Issue by 64kramsystem** [immunant/c2rust#611]
It'd be helpful if `c2rust transpile --binary` raised an error, if the parameter provided is incorrect.

Currently, the binary functionality fails silently, and if the user accidentally typoes the name, the result is very confusing.

**Comment by kkysen** [comment]
Hi @64kramsystem!  Could you give me a reproducible example so we can look into this more?

**Comment by 64kramsystem** [comment]
> Hi @64kramsystem! Could you give me a reproducible example so we can look into this more?

Sure! Instructions follow.

```sh
git clone git@github.com:Blzut3/CatacombSDL.git
cd Cata*/
cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1
c2rust transpile --binary pizzarulez compile_commands.json
cat Cargo.toml
```

This is the result; it's missing the `[[bin]]` section:

```toml
[workspace]
members = [
]
[package]
name = "c2rust_out"
authors = ["C2Rust"]
version = "0.0.0"
publish = false
edition = "2018"
autobins = false

[lib]
name = "c2rust_out"
path = "c2rust-lib.rs"
crate-type = ["staticlib", "rlib"]
[dependencies]
libc= "0.2"
```

If I set the correct name:

```sh
git clean -f
c2rust transpile --binary catacomb compile_commands.json
cat Cargo.toml
```

The `[[bin]]` section is generated:

```toml
[workspace]
members = [
]
[package]
name = "c2rust_out"
authors = ["C2Rust"]
version = "0.0.0"
publish = false
edition = "2018"
autobins = false

[lib]
name = "c2rust_out"
path = "c2rust-lib.rs"
crate-type = ["staticlib", "rlib"]
[[bin]]
path = "catacomb.rs"
name = "catacomb"
[dependencies]
libc= "0.2"
```

**Comment by kkysen** [comment]
Thanks for the instructions!  The reason why there's no error is because we iterate through the modules and check if they match the binaries you supply with `--binaries`.  I do agree we should turn this into an error, though, if the provided `--binaries` don't match any modules seen, or at the very least a warning.

**Comment by kkysen** [comment]
@64kramsystem, would it be good enough if a warning were printed if a supplied `--binary` was not used for any module?  And then a list of the candidate modules?  This would be done at the end of transpilation.

**Comment by 64kramsystem** [comment]
> @64kramsystem, would it be good enough if a warning were printed if a supplied `--binary` was not used for any module? And then a list of the candidate modules? This would be done at the end of transpilation.

Thanks for checking! I think the core value is to bring attention to the problem to the user. The list of modules is a nice addition; in my specific case, just the warning would have been enough :smile: but it definitely depends on the target audience.
